test/api_compatibility.js

```diff
@@ -103,30 +96,78 @@ module.exports = function apiCompatibilityChecks(_createTracer) {
+                    expect(span[name]).to.be.a('function');
+                });
+            });
+
+            it ('extract returns span context with allowed functions', function() {
+                // pass by default if no factory map was specified.
+                if (carrierFormatFactoryMap) {
+                    var carrierFactory = carrierFormatFactoryMap[opentracing.FORMAT_TEXT_MAP];
+                    tracer.extract(opentracing.FORMAT_TEXT_MAP, carrierFactory());
+                }
             });
         });
 
+        describe('Span', function() {
+            _.each(spanFunctions, function(name) {
+                it(name + ' should be a method', function() {
+                    expect(span[name]).to.be.a('function');
+                });
+            });
 
-        describe('Reference', function() {
-            var tracer = createTracer();
-            var span = tracer.startSpan('test_span');
-            var ref = new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, span.context());
+            var spanExecutions = [
+                {name: 'tracer', args: []},
+                {name: 'context', args: []},
+                {name: 'setOperationName', args: ['name']},
+                {name: 'setTag', args: ['key', 'value']},
+                {name: 'addTags', args: [{'key': 'value'}]},
+                {name: 'log', args: [
+                    {'event': 'event-name', 'payload': {'key': 'value'}}
+                ]},
+                {name: 'logEvent', args: ['eventName', null]},
+                {name: 'logEvent', args: ['eventName', {'key': 'value'}]},
+                {name: 'finish', args: []},
+                {name: 'setBaggageItem', args: ['key', 'value']},
+                {name: 'getBaggageItem', args: ['key']},
+            ];
+            _.each(spanExecutions, function(a) {
+                it(a.name + 'should not throw exceptions', function() {
+                    span[a.name].apply(span, a.args);
+                })
+            });
 
-            it('should be a class', function() {
-                expect(ref).to.be.an('object');
+            it('should set baggage and retrieve baggage', function() {
+                span.setBaggageItem('some-key', 'some-value');
+                var val = span.getBaggageItem('some-key');
+                if (checkBaggageValues) {
+                    assert.equal('some-value', val);
+                }
             });
 
-            it('should handle Spans and SpanContexts', function() {
-                var tracer = new opentracing.Tracer();
-                var span = tracer.startSpan('test_operation');
-                expect(function() { new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, span); }).to.not.throw(Error);
+            it('setTag should be chainable', function() {
+                var newSpan = span.setTag('key', 'value');
+                expect(span['setTag']).to.be.a('function');
```

Comment:
I would add a boolean `chainable` to `spanExecutions` table above, and test across the board, since setTag is not the only chainable method (logs and setOperationName should be too)